octobot_trading/personal_data/orders/order.py

```diff
@@ -477,41 +477,30 @@ def is_short(self):
         return self.side is enums.TradeOrderSide.SELL
 
     def update_from_raw(self, raw_order):
-        if self.side is None or self.order_type is None:
-            try:
+        try:
+            if self.side is None or self.order_type is None:
                 self._update_type_from_raw(raw_order)
-                if self.taker_or_maker is None:
-                    self._update_taker_maker()
-            except KeyError:
-                logging.get_logger(self.__class__.__name__).warning("Failed to parse order side and type")
-
-        price = raw_order.get(enums.ExchangeConstantsOrderColumns.PRICE.value, 0.0) or 0.0
-        # TODO replace with := when cython will be supporting it
-        stop_price = raw_order.get(enums.ExchangeConstantsOrderColumns.STOP_PRICE.value, None)
-        if stop_price is not None and (price is None or price == 0):
-            # parse stop price when available
-            price = stop_price
-        filled_price = decimal.Decimal(str(price))
-        # set average price with real average price if available, use filled_price otherwise
-        average_price = decimal.Decimal(str(raw_order.get(enums.ExchangeConstantsOrderColumns.AVERAGE.value, 0.0)
-                                            or filled_price))
-
-        return self.update(
-            symbol=str(raw_order.get(enums.ExchangeConstantsOrderColumns.SYMBOL.value, None)),
-            current_price=decimal.Decimal(str(price)),
```

Comment:
if we remove this, we gotta be 100% sure that parsers are returning decimals all the time

Reply:
yes parsers make sure its complete or raising all the time, once tests are done we should be good